connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MySQLIntegrationSuite.scala

```diff
@@ -241,6 +241,323 @@ class MySQLIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCTest
+          Seq(intColumns, longColumns, floatColumns, decimalColumns)
+        ),
+        generateTests(
+          "ATAN2",
+          "ABS(ATAN2(COLUMN, 1) - ATAN2(VALUE, 1)) <= 0.00001",
+          Seq(intColumns, longColumns, floatColumns, decimalColumns)
+        ),
+        generateTests(
+          "DEGREES",
+          "ABS(DEGREES(COLUMN) - DEGREES(VALUE)) <= 0.00001",
+          Seq(intColumns, longColumns, floatColumns, decimalColumns)
+        ),
+        generateTests(
+          "RADIANS",
+          "ABS(RADIANS(COLUMN) - RADIANS(VALUE)) <= 0.00001",
+          Seq(intColumns, longColumns, floatColumns, decimalColumns)
+        ),
+        generateTests(
+          "SIGN",
+          "ABS(SIGN(COLUMN) - SIGN(VALUE)) <= 0.00001",
+          Seq(intColumns, longColumns, floatColumns, decimalColumns)
+        ),
+        generateTests(
+          "UPPER",
+          "UPPER(COLUMN) = VALUE",
+          Seq(
+            intColumns,
+            longColumns,
+            floatColumns,
+            strColumns,
+            decimalColumns
+          ),
+          Some(toStringLiteral)
+        ),
+        generateTests(
+          "LOWER",
+          "LOWER(COLUMN) = VALUE",
+          Seq(
+            intColumns,
+            longColumns,
+            floatColumns,
+            strColumns,
+            decimalColumns
+          ),
+          Some(toStringLiteral)
+        ),
+        generateTests(
+          "SHA1",
+          "SHA1(COLUMN) = SHA1(VALUE)",
+          Seq(strColumns, binaryColumns)
+        ),
+        generateTests(
+          "SHA2",
+          "SHA2(COLUMN, 256) = SHA2(VALUE, 256)",
+          Seq(strColumns, binaryColumns)
+        ),
+        generateTests(
+          "MD5",
+          "MD5(COLUMN) = MD5(VALUE)",
+          Seq(strColumns, binaryColumns)
+        ),
+        generateTests(
+          "CRC32",
+          "CRC32(COLUMN) = CRC32(VALUE)",
+          Seq(strColumns, binaryColumns)
+        ),
+        generateTests(
+          "BIT_LENGTH",
+          "BIT_LENGTH(COLUMN) = BIT_LENGTH(VALUE)",
+          Seq(
+            intColumns,
+            longColumns,
+            floatColumns,
+            strColumns,
+            binaryColumns,
+            decimalColumns
+          ),
+          Some(toStringLiteral)
+        ),
+        generateTests(
+          "CHAR_LENGTH",
+          "CHAR_LENGTH(COLUMN) = CHAR_LENGTH(VALUE)",
+          Seq(
+            intColumns,
+            longColumns,
+            floatColumns,
+            strColumns,
+            binaryColumns,
+            decimalColumns
+          ),
+          Some(toStringLiteral)
+        ),
+        generateTests(
+          "CONCAT",
+          "CONCAT(COLUMN) = VALUE",
+          Seq(
+            intColumns,
+            longColumns,
+            floatColumns,
+            strColumns,
+            binaryColumns,
+            decimalColumns
+          ),
+          Some(toStringLiteral)
+        )
+      ).flatten
+
```

Comment:
Please the test cases for cast here.

Reply:
But, I will add other cast tests to changed types as well. This will be pretty quick.

Reply:
Added new tests for cast, PTAL

Reply:
Moved to #49453